### ISSUE TITLE
New Rule: no-computed-without-dependent-keys

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/eslint-plugin-ember#282]
The proposed rule should warn about computed properties without any dependent keys:
```js
foo: computed(function() {
  return 42;
})
```

instead the property should be initialized in the constructor:
```js
init() {
  this._super(...arguments);
  this.set('foo', 42);
}
```

**Comment by Turbo87** [comment]
Yeah, I guess in those situations it can make sense. That point should probably be added to the rule documentation then.

**Comment by Kerrick** [comment]
We've found `foo: computed(() => ({ some: 'stuff', goes: 'here' }),` to be a helpful pattern to avoid ember/avoid-leaking-state-in-ember-objects when we want to have a hash accessible by both the JS and the template. Would the preferred pattern _really_ be to set that in `init`? It would be harder to find while scanning the component's JS looking for a property you use in the template.

**Comment by Turbo87** [comment]
@Kerrick yes, putting it in `init` is usually better in these cases
